sympy/physics/units/tests/test_quantities.py

```diff
@@ -239,10 +241,30 @@ def test_sum_of_incompatible_quantities():
 
 
 def test_factor_and_dimension():
+    assert (3000, Dimension(1)) == Quantity._collect_factor_and_dimension(3000)
     assert (1001, length) == Quantity._collect_factor_and_dimension(meter + km)
     assert (2, length/time) == Quantity._collect_factor_and_dimension(
         meter/second + 36*km/(10*hour))
 
     x, y = symbols('x y')
     assert (x + y/100, length) == Quantity._collect_factor_and_dimension(
         x*m + y*centimeter)
+
+    cH = Quantity('cH', amount_of_substance/volume)
+    pH = -log(cH)
+
+    assert (1, volume/amount_of_substance) == Quantity._collect_factor_and_dimension(
+        exp(pH))
+
+    v_w1 = Quantity('v_w1', length/time, 1.5*meter/second)
```

Comment:
in SymPy, it's usually better to represent floating point numbers as rationals, e.g. `Rational(3, 2)`.